src/osgPlugins/obj/ReaderWriterOBJ.cpp

```diff
@@ -243,6 +243,9 @@ static void load_material_texture(    obj::Model &model,
         {
             // first try with database path of parent.
             image = osgDB::readRefImageFile(model.getDatabasePath()+'/'+filename, options);
+
+			// Preserve relative path in case filename is written out later
+			image->setFileName(filename);
```

Comment:
You may want to change the tab characters in your indents to spaces to be consistent with the existing code.

Reply:
Fixed